File: src/frontend/src/app/core/services/candidato.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { Candidato, CreateCandidato, UpdateCandidato } from '../models';

@Injectable({
  providedIn: 'root'
})
export class CandidatoService {
  private endpoint = '/candidatos';

  constructor(private api: ApiService) {}

  getAll() {
    return this.api.get<Candidato[]>(this.endpoint);
  }

  getById(id: number) {
    return this.api.get<Candidato>(`${this.endpoint}/${id}`);
  }

  getByProcessoId(processoId: number) {
    return this.api.get<Candidato[]>(`${this.endpoint}/processo/${processoId}`);
  }

  getPontuacao(id: number) {
    return this.api.get<number>(`${this.endpoint}/${id}/pontuacao`);
  }

  create(data: CreateCandidato) {
    return this.api.post<Candidato>(this.endpoint, data);
  }

  update(id: number, data: UpdateCandidato) {
    return this.api.put<Candidato>(`${this.endpoint}/${id}`, data);
  }

  delete(id: number) {
    return this.api.delete<void>(`${this.endpoint}/${id}`);
  }
}

File: src/frontend/ProcessoSelecao.Blazor/Services/CandidatoService.cs
using ProcessoSelecao.Blazor.Models;

namespace ProcessoSelecao.Blazor.Services;

public class CandidatoService
{
    private readonly ApiService _api;
    private const string Endpoint = "candidatos";

    public CandidatoService(ApiService api)
    {
        _api = api;
    }

    public virtual async Task<List<Candidato>> GetAllAsync()
        => await _api.GetAsync<List<Candidato>>(Endpoint) ?? new();

    public virtual async Task<Candidato?> GetByIdAsync(long id)
        => await _api.GetAsync<Candidato>($"{Endpoint}/{id}");

    public virtual async Task<List<Candidato>> GetByProcessoIdAsync(long processoId)
        => await _api.GetAsync<List<Candidato>>($"{Endpoint}/processo/{processoId}") ?? new();

    public virtual async Task<double> GetPontuacaoAsync(long id)
        => await _api.GetAsync<double>($"{Endpoint}/{id}/pontuacao");

    public virtual async Task<Candidato?> CreateAsync(CreateCandidato dto)
        => await _api.PostAsync<Candidato>(Endpoint, dto);

    public virtual async Task<Candidato?> UpdateAsync(long id, UpdateCandidato dto)
        => await _api.PutAsync<Candidato>($"{Endpoint}/{id}", dto);

    public virtual async Task DeleteAsync(long id)
        => await _api.DeleteAsync($"{Endpoint}/{id}");
}
